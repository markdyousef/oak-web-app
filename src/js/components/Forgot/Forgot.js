// @flow
import React, { Component, PropTypes } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router';
import Input from '../shared/Input';
import colors from '../../styles/colors';
import Arrow from '../../icons/rightArrow';

const Container = styled.section`
    width: 100%;
`;

const Box = styled.div`
    margin: auto;
    margin-top: 125px;
    width: 480px;
    ${''/* height: 380px; */}
    background-color: ${colors.white};
    padding: 45px 100px;
    display: flex;
    flex-direction: column;
    border: 1px solid ${colors.lightGrey};
    border-radius: 3px;
    & h1 {
        font-size: 32px;
        font-weight: bold;
        text-align: center;
    }
    & div {
        margin-top: 15px;
    }
`;

const Button = styled.button`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    background-color: ${colors.green};
    border-radius: 3px;
    border: 1px solid ${colors.green};
    color: ${colors.white};
    font-size: 15px;
    font-weight: bold;
    font-family: 'Proxima Nova';
    padding: 12px;
    cursor: pointer;
    &:hover {
        box-shadow: inset 0 -2px rgba(0, 0, 0, 0.15);
        transition: 0.2s;
    }
    &:after {
        box-shadow: inset 0 -2px rgba(0, 0, 0, 0.15);
        transition: 0.2s;
    }
    & svg {
        fill: ${colors.white};
        margin-left: 5px;
    }
`;


const ErrorMessage = styled.div`
    width: 100%;
    background-color: ${colors.orange};
    border-radius: 3px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    color: ${colors.white};
`;

const Forgot = styled.div`
    max-width: 250px;
    margin: auto;
    margin-top: 32px;
    color: ${colors.grey};
    & a {
        font-weight: bold;
        color: ${colors.grey};
    }
`;

class Login extends Component {
    static propTypes = {
        resetPassword: PropTypes.func.isRequired,
        router: PropTypes.shape({
            replace: PropTypes.func
        }).isRequired
    };
    constructor() {
        super();
        this.onSubmit = this.onSubmit.bind(this);
        this.state = {
            email: '',
            message: null
        };
    }
    onSubmit() {
        const { resetPassword, router } = this.props;
        const { email } = this.state;

        // clear messsage
        this.setState({ message: null });
        return resetPassword(email)
            .then(res => console.log(res))
            .catch(err => console.log(err))
    }
    render() {
        const { email, password, message } = this.state;
        const { router } = this.props;
        return (
            <Container>
                <Box>
                    <h1>Reset password</h1>
                    <div>
                        <Input
                            title="EMAIL"
                            value={email}
                            onChange={value => this.setState({ email: value, message: null })}
                            type="email"
                            placeholder="Email"
                            onKeyDown={(e) => {
                                if (e.keyCode === 13) {
                                    this.onSubmit();
                                }
                            }}
                        />
                    </div>
                    <div>
                        <Button
                            onClick={this.onSubmit}
                        >
                            Reset password
                            <Arrow />
                        </Button>
                    </div>
                    <div>
                        {message &&
                            <ErrorMessage>
                                <h5>{message}</h5>
                            </ErrorMessage>
                        }
                    </div>
                </Box>
                <Forgot>
                    <p>Remember your password? <Link to="/login">Sign in!</Link></p>
                </Forgot>
            </Container>
        );
    }
}

export default Login;
