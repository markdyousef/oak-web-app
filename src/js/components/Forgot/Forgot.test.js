import React from 'react';
import renderer from 'react-test-renderer';
import Forgot from './Forgot';

it('renders correctly', () => {
    const props = {
        resetPassword: () => {},
        router: {}
    }
    const component = renderer.create(
        <Forgot {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
