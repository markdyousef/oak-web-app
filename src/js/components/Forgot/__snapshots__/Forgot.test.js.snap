exports[`test renders correctly 1`] = `
<section
  className="gmuKOB">
  <div
    className="khSIyb">
    <h1>
      Reset password
    </h1>
    <div>
      <div>
        <h5
          className="jIUOlj">
          EMAIL
        </h5>
        <input
          className="fhluRP"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Email"
          type="email"
          value="" />
      </div>
    </div>
    <div>
      <button
        className="leFGPx"
        onClick={[Function]}>
        Reset password
        <svg
          height="16"
          viewBox="0 0 16 16"
          width="16">
          <g>
            <path
              d="M10.9,2.8l4.8,4.8c0.1,0.1,0.1,0.2,0.1,0.4s-0.1,0.3-0.1,0.4l-4.8,4.8c-0.2,0.2-0.5,0.2-0.7,0s-0.2-0.5,0-0.7l4-4H0.6 C0.3,8.5,0.1,8.3,0.1,8s0.2-0.5,0.5-0.5h13.6l-4-4C10,3.3,10,3,10.2,2.8S10.7,2.6,10.9,2.8z" />
          </g>
        </svg>
      </button>
    </div>
    <div />
  </div>
  <div
    className="lbNTMF">
    <p>
      Did you remember your old password? 
      <a
        onClick={[Function]}
        style={Object {}}>
        Sign in!
      </a>
    </p>
  </div>
</section>
`;
